pkg/infra/game/decision/rule.go

```diff
@@ -0,0 +1,22 @@
+package decision
```

Comment:
Any reason why rules are moved to decision? Maybe a sub package under decision might make more sense, if it needs to be under this namespace.

Reply:
Yes 2 amazing reasons - check these out 👀: 
1. It's the first place I could find to put them to avoid cyclic imports after around 15 mins or so of moving it all around
2. Conceptually not too compelling that it should be that tightly coupled to messages (maybe before yes but now that this is being used in more than 1 place it should be a touch more exposed)

But yh do agree that it should be a bit more distinct than currently in this pr